src/Bladeburner/Actions/TeamCasualties.ts

```diff
@@ -0,0 +1,49 @@
+import type { getRandomIntInclusive } from "../../utils/helpers/getRandomIntInclusive";
+
+export enum CasualtyFactor {
+  LOW_CASUALTIES = 0.5, // 50%
+  HIGH_CASUALTIES = 1, // 100%
+}
+
+export interface OperationTeam {
+  teamSize: number;
+  teamLost: number;
+
+  killSupportingSleeves(sleeveDeaths: number): void;
+}
+
+/**
+ * Some actions (Operations and Black Operations) use teams for success bonus
+ * and may result in casualties, reducing the player's hp, killing team members
+ * and killing sleeves (to shock them, sleeves are immortal)
+ *
+ * Caveats: TeamSize = Human Team + Supporting Sleeves set to this action
+ */
+export class TeamCasualties {
+  constructor(private severity: CasualtyFactor, private teamSizeUsed: number, private supportingSleeves: number) {}
+
+  rollOutcome(random: typeof getRandomIntInclusive, team: OperationTeam) {
+    const deaths = random(this.bestCase, this.worstCase);
+    const humans = this.teamSizeUsed - this.supportingSleeves;
+    const humanDeaths = Math.min(humans, deaths);
+    const damagedSleeves = deaths - humanDeaths;
```

Comment:
You should double-check the logic here.

`teamSizeUsed` = `action.teamCount` and `supportingSleeves` = `this.sleeveSize`. Let's say we have 4 team members (from recruitment), 2 supporting sleeves and we set team count to 4 when doing the operation: `teamSizeUsed` = 4, `supportingSleeves` = 2, and `this.teamSize` = 6.

If `deaths` = 4, `damagedSleeves` should be 0, but, in your implementation, `damagedSleeves` is 2 (`deaths - Math.min(teamSizeUsed - supportingSleeves, deaths)` = 4 - min(4 - 2, 2) = 2.

In the current implementation, `damagedSleeves` is calculated based on `this.teamSize`, not `action.teamCount`. It's `this.teamSize - losses - this.sleeveSize`.

The current logic: when there is danger, we use team members (from recruitment) as cannon fodder and let them die before our sleeves. If you want to change this logic, you need to state that in the PR, because it's a balance change.

Reply:
I haven't considered killing off human team members that did not partake in the action. I will check with the current dev branch and have that behavior match. I can't confirm whether you're correct outside of that.

Reply:
Checked dev, team members not present also get killed off. Will restore original behavior. As d0s mentioned the fix can be a one-liner as a follow-up, plus the test.